Extensions/ShareTo/InitialViewController.swift

```diff
@@ -66,14 +66,13 @@ class InitialViewController: UIViewController, ShareControllerDelegate {
             if destinations.contains(ShareDestinationReadingList) {
                 profile.readingList?.createRecordWithURL(item.url, title: item.title ?? "", addedBy: UIDevice.current.name)
             }
-            
+
             if destinations.contains(ShareDestinationBookmarks) {
-                profile.bookmarks.shareItem(item)
+                profile.bookmarks.shareItem(item) >>> {
+                    profile.shutdown()
+                    self.extensionContext!.completeRequest(returningItems: [], completionHandler: nil)
+                }
             }
-
```

Comment:
We might want to move this into an `else` block so if we fail the `if` check above we always make sure to shutdown and call `completeRequest`.